search-android/src/main/java/uk/os/elements/search/android/app/SearchActivity.java

```diff
@@ -0,0 +1,639 @@
+        searchProviders.add(new GridReferenceProvider());
+        searchProviders.add(new LatLonProvider());
+        searchProviders.add(new LocalDatabaseProvider(this));
+        return searchProviders;
+    }
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setTheme(R.style.SearchTheme);
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.search_activity);
+
+        mRecentsManager = getRecentsManager();
+        mSearchManager = getSearchManager(mRecentsManager);
+
+        mRecyclerView = (RecyclerView) findViewById(R.id.search_results);
+        mSearchView = (EditText) findViewById(R.id.search_searchview);
+        mToolbar = (Toolbar) findViewById(R.id.search_toolbar);
+        mClear = (ImageButton) findViewById(R.id.search_clear);
+        mMic = (ImageButton) findViewById(R.id.search_mic);
+        mMicHollow = (ImageButton) findViewById(R.id.search_mic_hollow);
+        mRecents = findViewById(R.id.search_recents);
+        mErrorRecents = findViewById(R.id.search_error_recents);
+        mActiveLine = findViewById(R.id.search_searchview_line);
+        mEmptyLine = findViewById(R.id.search_searchview_line_empty);
+        mErrorNetwork = findViewById(R.id.search_error_network);
+        mErrorHttp = findViewById(R.id.search_error_http);
+        mErrorConversion = findViewById(R.id.search_error_conversion);
+        mNoSearchResults = findViewById(R.id.search_no_results);
+        mErrorCatchAll = findViewById(R.id.search_error_catch_all);
+
+        mSearchView.setOnFocusChangeListener(new View.OnFocusChangeListener() {
+            @Override
+            public void onFocusChange(View v, boolean hasFocus) {
+                updateSearchViewState();
+            }
+        });
+
+        View.OnClickListener l = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Intent intent = new Intent(RecognizerIntent.ACTION_RECOGNIZE_SPEECH);
+                intent.putExtra(RecognizerIntent.EXTRA_LANGUAGE_MODEL,
+                        RecognizerIntent.LANGUAGE_MODEL_FREE_FORM);
+                intent.putExtra(RecognizerIntent.EXTRA_PROMPT, getString(R.string.search_voice_prompt));
+                startActivityForResult(intent, REQUEST_CODE);
+            }
+        };
+        mMic.setOnClickListener(l);
+        mMicHollow.setOnClickListener(l);
+
+        mSearchView.addTextChangedListener(new TextWatcher() {
+            @Override
+            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
+
+            }
+
+            @Override
+            public void onTextChanged(CharSequence text, int start, int before, int count) {
+                mQueryConcern.setSearchText(text.toString());
+                updateSearchViewState();
+            }
+
+            @Override
+            public void afterTextChanged(Editable s) {
+
+            }
```

Comment:
sadly not as those methods are abstract http://developer.android.com/reference/android/text/TextWatcher.html